src/cpp/desktop/DesktopSessionLauncher.cpp

```diff
@@ -212,6 +214,102 @@ void SessionLauncher::closeAllSatellites()
    }
 }
 
+Error getRecentSessionLogs(std::string* pLogFile, std::string *pLogContents)
+{
+   // Collect R session logs
+   std::vector<FilePath> logs;
+   Error error = userLogPath().children(&logs);
+   if (error)
+   {
+      return error;
+   }
+
+   // Sort by recency in case there are several session logs --
+   // inverse sort so most recent logs are first
+   std::sort(logs.begin(), logs.end(), [](FilePath a, FilePath b)
+   {
+      return a.lastWriteTime() < b.lastWriteTime();
+   });
+
+   // Loop over all the log files and stop when we find a session log
+   // (desktop logs are also in this folder)
+   for (const auto& log: logs)
+   {
+      if (log.filename().find("rsession") != std::string::npos)
```

Comment:
nit: should we only look for files starting with `"rsession-"`? (I could see `rsession` or `rsessions` being a plausible username)

Reply:
The only logs in that directory (that we write) are `rdesktop.log` and `rsession-username.log`. So it's just fine if we accidentally match `rsession` in the username. :-)